src/core/defaultappearance.js

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isName, Name } from "./primitives.js";
+import { escapePDFName } from "./core_utils.js";
+
+// Parse DA to extract font and color informations
+function parseDefaultAppearance(str) {
+  const result = {
+    fontSize: null,
+    fontName: null,
+    color: null,
+  };
+
+  if (!str) {
+    return result;
+  }
+
+  const stack = [];
+
+  for (const token of str.split(/\s+/)) {
```

Comment:
In practice it's not *always* guaranteed that operators are properly separated by whitespace, since bad PDF generators sometimes "glue" them together; see examples in these unit-tests:
https://github.com/mozilla/pdf.js/blob/master/test/unit/evaluator_spec.js

Again, re-using the `Parser`/`Lexer´ might help here!